src/nvme/fabrics.c

```diff
@@ -207,7 +207,7 @@ static int add_int_argument(char **argstr, char *tok, int arg, bool allow_zero)
 {
 	char *nstr;
 
-	if (arg < 0 || (!arg && !allow_zero))
+	if (!arg && !allow_zero)
```

Comment:
I wonder it it wouldn't be more sensible to introduce an 'add_uint_argument' function or an additional argument 'allow_negative'; negative values are invalid for most arguments, and with this change we lose the ability to check for negative values.

Reply:
At the moment, libnvme doesn't check the validity of the arguments passed by the user, it just filters them out, silently.
example: 
# ./nvme connect -t tcp -a 10.37.153.75 -n test-nqn -s 4080  --tos=-100
this command successfully completes even if tos is invalid.

I think it would be better to pass to the kernel whatever the user wants and let the kernel decide if it should accept, ignore or return an error.

Reply:
I would agree with you if the user would have a chance to figure out what went wrong when sending down the string.
ATM there is no programmatic way how the user (or application) could differentiate between invalid arguments and the connection failing due to controller errors.
And if we don't have that we have to ensure that we're not passing down invalid arguments.

Reply:
Ah, right, I see the problem.
So the build_options() function has to be rewritten, because right now nvme-cli doesn't check if an argument has a valid value or not, it adds it to the string it will pass to the kernel or it filters it out without warnings...